lib/tool/serp_api.rb

```diff
@@ -16,23 +14,33 @@ class SerpApi < Base
       Input should be a search query.
     DESC
 
+    def initialize
+      depends_on "google_search_results"
+      require "google_search_results"
+    end
+
+    # Executes Google Search and returns hash_results JSON
+    # @param input [String] search query
+    # @return [Hash] hash_results JSON
+
+    def self.execute_search(input:)
+      new.execute_search(input: input)
+    end
```

Comment:
I considered not adding, this but didn't want to require other tools to instantiate this tool to use it.